src/ImageArchive/Image.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * Class Image.
+ *
+ * @package P4\MasterTheme\ImageArchive
+ */
+
+namespace P4\MasterTheme\ImageArchive;
+
+use JsonSerializable;
+use P4\MasterTheme\Exception\UploadFailed;
+use RuntimeException;
+
+/**
+ * Entity for images returned by the media API.
+ */
+class Image implements JsonSerializable {
+	public const ARCHIVE_ID_META_KEY = 'gp_archive_id';
+
+	/**
+	 * Media Library System Identifier.
+	 *
+	 * @var $archive_id
+	 */
+	private $archive_id;
+
+	/**
+	 * Image Title.
+	 *
+	 * @var $title
+	 */
+	private $title;
+
+	/**
+	 * Image Caption.
+	 *
+	 * @var $caption
+	 */
+	private $caption;
+
+	/**
+	 * Image Credit.
+	 *
+	 * @var $credit
+	 */
+	private $credit;
+
+	/**
+	 * Image restrictions.
+	 *
+	 * @var $restrictions
+	 */
+	private $restrictions;
+
+	/**
+	 * The attachment id if the image is stored in WordPress db.
+	 *
+	 * @var $wordpress_id
+	 */
+	private $wordpress_id;
+
+	/**
+	 * The Original language title of attachment.
+	 *
+	 * @var $original_language_title
+	 */
+	private $original_language_title;
+
+	/**
+	 * @var string|null The original language description.
+	 */
+	private $original_language_description;
+
+	/**
+	 * @var ImageSize[] Different available sizes for this image.
+	 */
+	private $sizes = [];
+
+	/**
+	 * @var ImageSize The image size which is the original (i.e. largest).
+	 */
+	private $original;
+
+	/**
+	 * Image constructor. Private to force use of meaningfully named static creation functions. That's also were the
+	 * properties are set, to avoid having to declare the same parameters in both functions.
+	 */
+	private function __construct() {
+	}
+
+	/**
+	 * @param array $data The data returned by Orange Logic API.
+	 * @param array $images_in_wordpress Array with all images that are already in WP.
+	 *
+	 * @return static The created instance.
+	 */
+	public static function from_api_response( array $data, array $images_in_wordpress ): self {
+		$image               = new self();
+		$image->archive_id   = $data['SystemIdentifier'];
```

Comment:
any idea of those types ? we could typecast and document the types in property declaration

Reply:
The API documentation doesn't explicitly state the types, but we can figure them out and document them here :+1: maybe something we can do in a follow up PR?

Reply:
follow-up PR sounds good to me :)